R/roxygen2.R

```diff
@@ -14,3 +14,43 @@ roxygen_param_dataset <- function(expected_vars = NULL) {
   }
   return(dataset_text)
 }
+
+
+# function to properly document the by_vars argument including specifying the use of exprs():
+
+roxygen_param_by_vars <- function(type = "group", additional_dataset = NULL) {
+
+  if (type == "group") {
+    by_vars_text <- "A list of quoted expressions specifying variables to group by. \n \n"
+  }
+
+
+  by_vars_text <- paste0(
+    by_vars_text,
+    # "The variables must be a unique key of the selected observations. \n \n",
+    #
+
+    "The expressions should be quoted using the `exprs()` function from ",
+    "the rlang package.",
+    "e.g. exprs(USUBJID, VISIT) \n \n"
+
+  )
+
+  if (!is.null(additional_dataset)) {
+    by_vars_text <- paste0(
+      by_vars_text,
+      "The by variables must be a unique key of the selected observations in ",
+      enumerate(additional_dataset), ". \n \n",
+      "Variables from ", enumerate(additional_dataset),
+      " can be renamed by naming the element, I.e. \n",
+      "by_vars = exprs(<name in input dataset› = ‹name in additional dataset>), similar to the dplyr joins.\n \n"
+    )
+  }
```

Comment:
Good to know! 
I will update the function so that there are two flags when there is an additional dataset with the following default arguments: 
`unique = FALSE`
`rename = FALSE`

Would you be able to go through the functions when I'm done and check whether the flags are set correctly? 
If it is mentioned in the original description this is no problem, but sometimes it is not, e.g. in `derive_expected_records()`. 
Or am I missing something obvious?

Reply:
If renaming is not mentioned in the description, I would assume that it is not possible.

Reply:
I will assume the same for "unique".